src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureAnalyzer.cs

```diff
@@ -210,15 +263,22 @@ private static bool SymbolIsStaticAndAbstract(ISymbol symbol, ConcurrentDictiona
         {
             ISymbol symbol = context.Symbol;
 
-            // Don't report diagnostics on symbols that are marked as Preview
-            if (ProcessPreviewAttribute(symbol, requiresPreviewFeaturesSymbols, previewFeatureAttributeSymbol))
+            // Don't report diagnostics on symbols that are marked as Preview or if the ContainingType is Preview. We'll miss 2+ level nested type/method definitions, but that's ok. Only users who suppress the analyzer will run into this case.
+            if (ProcessPreviewAttribute(symbol, requiresPreviewFeaturesSymbols, previewFeatureAttributeSymbol) ||
+                (symbol is IMethodSymbol method && method.AssociatedSymbol != null && ProcessPreviewAttribute(method.AssociatedSymbol, requiresPreviewFeaturesSymbols, previewFeatureAttributeSymbol)))
+            {
+                return;
+            }
+
+            ISymbol containingSymbol = symbol.ContainingType;
+            if (containingSymbol != null && ProcessPreviewAttribute(containingSymbol, requiresPreviewFeaturesSymbols, previewFeatureAttributeSymbol))
             {
                 return;
             }
 
             var reportDiagnostic = symbol switch
             {
-                ITypeSymbol typeSymbol => ProcessTypeSymbolAttributes(typeSymbol, requiresPreviewFeaturesSymbols, requiresPreviewFeaturesSymbolsToUsageType, previewFeatureAttributeSymbol),
+                ITypeSymbol typeSymbol => ProcessTypeSymbolAttributes(typeSymbol, requiresPreviewFeaturesSymbols, previewFeatureAttributeSymbol),
                 IPropertySymbol or IMethodSymbol => ProcessPropertyOrMethodAttributes(symbol, requiresPreviewFeaturesSymbols, requiresPreviewFeaturesSymbolsToUsageType, virtualStaticsInInterfaces, previewFeatureAttributeSymbol),
```

Comment:
Ah my very first PR had similar names, and the feedback I got was to use "Process*" because `TypeSymbolReferencesPreviewFeatures` was also updating a dictionary within in => it had a side effect that is not obvious from the name. I personally prefer the non `Process*` names, so I'll take your suggestion

Reply:
Cool; thanks. I think it's OK for verb-named methods to have side-effects (building caches essentially).